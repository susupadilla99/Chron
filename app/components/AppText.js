import React from 'react';
import { first } from 'react-native'


function AppText(props) {
  return (
    <div>
      
    </div>
  );
}

const styles = StyleSheet.create({
  
})

export default AppText;