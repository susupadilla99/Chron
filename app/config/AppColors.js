export default {
    black: '#000000',
    white: '#FFFFFF',
    backgroundColor: '#F8F8F8',
    lightBlue: '#9CEAE7',
    darkBlue: '#4FC0FF',
    veryDarkBlue: '#0075FF',
    blueGradient: ['#9CEAE7', '#4FC0FF'],
    gray: '#EAEAEA',
    darkGray: '#B1B1B1',
};