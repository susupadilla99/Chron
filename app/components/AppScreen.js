import React from 'react';
import { SafeAreaView, StyleSheet, StatusBar } from 'react-native';

import AppColors from '../config/AppColors';

function AppScreen({children}) {
  return (
    <SafeAreaView style={styles.screen}>
      <StatusBar
        animated={true}
        backgroundColor={AppColors.backgroundColor}
        barStyle='dark-content' />
      {children}
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    backgroundColor: AppColors.backgroundColor,
    justifyContent: 'center',
    alignItems: 'center',
  },
})

export default AppScreen;