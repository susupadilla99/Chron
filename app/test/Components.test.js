import React from 'react';
import renderer from 'react-test-renderer';

import AppScreen from '../components/AppScreen';
import AppText from '../components/AppText';
import AppColors from '../config/AppColors';

test('AppText', () => {
  // AppText renders
  var json = renderer.create(<AppText/>).toJSON();
  expect(json).toBeDefined();

  // AppText has correct attributes
  var json = renderer.create(
    <AppText
      size={30}
      color={AppColors.gray}
      style={{textDecorationLine: 'underline'}} 
      >Hi mom!</AppText>).toJSON();
  expect(json.children[0]).toBe("Hi mom!");
  expect(json.props.style[0]).toEqual({
    color: AppColors.black,
    fontFamily: 'Avenir',
    fontSize: 20,
  });
  expect(json.props.style[1]).toEqual({fontSize: 30, color: AppColors.gray});
  expect(json.props.style[2]).toEqual({textDecorationLine: 'underline'});
  expect(json).toMatchSnapshot();
});

test('AppScreen', () => {
  // AppScreen renders
  var json = renderer.create(<AppScreen/>).toJSON();
  expect(json).toBeDefined();

  // AppScreen displays childrens correctly
  var textComponent = renderer.create(<AppText>Hi mom!</AppText>).toJSON();
  var json = renderer.create(
    <AppScreen>
      <AppText>Hi mom!</AppText>
    </AppScreen>).toJSON();
  expect(json.props.style.backgroundColor).toEqual(AppColors.backgroundColor);
  expect(json.children[0].children[0]).toEqual(textComponent);
  expect(json).toMatchSnapshot();
})