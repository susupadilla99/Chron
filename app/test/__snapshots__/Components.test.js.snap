// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F8F8",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "flex": 1,
        "flexDirection": "column",
        "marginTop": undefined,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Avenir",
            "fontSize": 20,
          },
          Object {
            "color": undefined,
            "fontSize": undefined,
          },
          undefined,
        ]
      }
    >
      Hi mom!
    </Text>
  </View>
</View>
`;

exports[`AppText 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "Avenir",
        "fontSize": 20,
      },
      Object {
        "color": "#EAEAEA",
        "fontSize": 30,
      },
      Object {
        "textDecorationLine": "underline",
      },
    ]
  }
>
  Hi mom!
</Text>
`;
